File: Pset's/pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
#include <string.h>
#include <ctype.h>

int main(void)
{
    // Promping user to gives us text
    string text = get_string("Text: ");
    
    // setting base value for our integers
    int letters = 0;
    int words = 1;
    int sentences = 0;
    
    // making for-loop to calculate end values for our integers posted above
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
        
        if (isspace(text[i]))
        {
            words++;
        }
        
        if (text[i] == '!' || text[i] == '?' || text[i] == '.')
        {
            sentences++;
        }
    }
    
    // calculating index per given formula using our collected data above
    float L = (letters / (float) words) * 100;
    float S = (sentences / (float) words) * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    
    // printing results per grade level
    if (index <= 0)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: Pset's/pset6/readability/readability.py
# importing function from cs50 lib
from cs50 import get_string

# Promping user to gives us text
text = get_string("Text: ")

# setting base value
letters = 0
words = 1
sentences = 0

# making for-loop to calculate end values
for i in range(len(text)):
    if text[i].isalpha():
        letters += 1
    if text[i].isspace():
        words += 1
    if (text[i] == '!' or text[i] == '?' or text[i] == '.'):
        sentences += 1
    
# calculating index per given formula using our collected data above
L = (letters / words) * 100
S = (sentences / words) * 100
index = round(0.0588 * L - 0.296 * S - 15.8)

# printing rsults
if (index <= 0):
    print("Before Grade 1")
elif (index >= 16):
    print("Grade 16+")
else:
    print(f"Grade {index}")

